tv_apps/tv_shared/js/card_manager.js

```diff
@@ -577,21 +601,20 @@
       });
     },
 
-    // TODO: need to be protected by state and _reloadCardList
     getCardList: function cm_getCardList() {
       var that = this;
       return this._getPipedPromise('getCardList', function(resolve, reject) {
-        if (!that._isCardListLoaded()) {
-          that._reloadCardList().then(function() {
-            resolve(that._cardList);
-          });
-        } else {
-          resolve(that._cardList);
-        }
+        that._asyncSemaphore.wait(function() {
+          resolve();
```

Comment:
you can combine the `that._reloadCardList()` below here as `resolve(that._reloadCardList())`.